src/main/java/org/graylog/integrations/okta/OktaResource.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.integrations.okta;
+
+import com.codahale.metrics.annotation.Timed;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.graylog2.plugin.rest.PluginRestResource;
+import org.graylog2.rest.resources.system.inputs.AbstractInputsResource;
+import org.graylog2.shared.inputs.MessageInputFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Api(value = "Okta", description = "Okta Integrations")
+@Path("/okta")
+@Produces(MediaType.APPLICATION_JSON)
+public class OktaResource extends AbstractInputsResource implements PluginRestResource {
```

Comment:
I am not up-to-speed on the requirements, so just a question regarding that: Are both the Okta Input and the API endpoint needed?

Reply:
yes. API endpoint for okta is to reach okta syslogs thru graylog API browser.  Okta Input is for Http polling.